src/vsr/replica.zig

```diff
@@ -1388,6 +1395,18 @@ pub fn ReplicaType(
                 return;
             }
 
+            switch (self.syncing) {
+                .idle => {},
+                .canceling_commit, .canceling_grid => {
+                    // Ignore further messages until finishing (asynchronous) processing of sync SV.
+                    // Notably, this prevents our view from jumping ahead of SV.
+                    assert(self.sync_start_view != null);
+                    log.warn("{}: on_message: ignoring (syncing)", .{self.replica});
+                    return;
+                },
+                .updating_checkpoint => {},
+            }
```

Comment:
TBH, this feels like a cludge to me, as it violates "embrace concurrency" principle. But also, a replica in the middle of `on_start_view` is in a rather weird state, and carefully handling that was hard.